Tests/tests/test_FrontPanel.cpp

```diff
@@ -0,0 +1,820 @@
+        Plugin::initDone = 0;
+        Plugin::fpIndicators.clear();
+        IarmBus::getInstance().impl = nullptr;
+    }
+};
+
+class FrontPanelInitializedEventTest : public FrontPanelInitializedTest {
+protected:
+    ServiceMock service;
+    Core::JSONRPC::Message message;
+    FactoriesImplementation factoriesImplementation;
+    PluginHost::IDispatcher* dispatcher;
+
+    FrontPanelInitializedEventTest()
+        : FrontPanelInitializedTest()
+    {
+        PluginHost::IFactories::Assign(&factoriesImplementation);
+
+        dispatcher = static_cast<PluginHost::IDispatcher*>(
+            plugin->QueryInterface(PluginHost::IDispatcher::ID));
+        dispatcher->Activate(&service);
+    }
+
+    virtual ~FrontPanelInitializedEventTest() override
+    {
+        dispatcher->Deactivate();
+        dispatcher->Release();
+
+        PluginHost::IFactories::Assign(nullptr);
+    }
+};
+
+class FrontPanelInitializedEventDsTest : public FrontPanelInitializedEventTest {
+protected:
+
+    ColorMock colorImplMock;
+
+    FrontPanelInitializedEventDsTest()
+        : FrontPanelInitializedEventTest()
+    {
+        device::FrontPanelIndicator::Color::getInstance().impl = &colorImplMock;   
+    }
+
+    virtual ~FrontPanelInitializedEventDsTest() override
+    {
+        device::FrontPanelIndicator::Color::getInstance().impl = nullptr;
+    }
+};
+
+TEST_F(FrontPanelTest, RegisteredMethods)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("powerLedOn")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("powerLedOff")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setClockBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getClockBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getFrontPanelLights")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getPreferences")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setPreferences")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setLED")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setBlink")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("set24HourClock")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("is24HourClock")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setClockTestPattern")));
+}
+
+TEST_F(FrontPanelInitializedEventDsTest, setBrightnessWIndex)
+{
+    //device::FrontPanelIndicator panel;
+    testing::NiceMock<FrontPanelIndicatorMock> indicatorMock;
+    device::FrontPanelIndicator::getInstance().impl = &indicatorMock;
+
+
+    ON_CALL(indicatorMock, getInstanceString)
+        .WillByDefault(::testing::Invoke(
+            [&](const std::string& name) -> device::FrontPanelIndicator& {
+                EXPECT_EQ("Power", name);
+                return device::FrontPanelIndicator::getInstance();
```

Comment:
it's hard to understand
`impl` is set several times
is it indicatorMock, or frontPanelIndicatorImplMock?
name "FrontPanelInitializedEventDsTest" doesn't explain it
can getInstanceString return a local object set up here?

Reply:
Updated to make it more concise and more consistent with the mock that is being used.